File: java/src/main/java/br/com/vps/consulting/b2b/management/partner/infrastructure/api/request/CreatePartnerRequest.java
package br.com.vps.consulting.b2b.management.partner.infrastructure.api.request;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

import java.math.BigDecimal;

public record CreatePartnerRequest(
        @Schema(description = "Nome do parceiro")
        @NotBlank String name,

        @Schema(description = "Documento do parceiro (CPF ou CNPJ)")
        @NotBlank String document,

        @Schema(description = "Limite de crédito inicial do parceiro", example = "10000.00")
        @NotNull @Positive BigDecimal creditLimit
) {}

File: kotlin/src/main/kotlin/br/com/vps/consulting/b2b/management/partner/infrastructure/api/request/CreatePartnerRequest.kt
package br.com.vps.consulting.b2b.management.partner.infrastructure.api.request

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank

data class CreatePartnerRequest(
    @field:Schema(description = "Nome do parceiro")
    @field:NotBlank(message = "O nome não pode estar em branco")
    val name: String,

    @field:Schema(description = "Documento do parceiro (CPF ou CNPJ)")
    @field:NotBlank(message = "O documento não pode estar em branco")
    val document: String,
)
